### ISSUE TITLE
Partial or complete text disappearing when trailing space is present.

### ISSUE THREAD

**Issue by andrewleek-droplab** [mimshwright/pixi-tagged-text#118]
First, thanks for a great library!

I came across this bug when ingesting a string that happened to have a trailing space.  I was working with word wrapping, so initially I thought it only applied to word wrapping, but I have since been able to reproduce it when wordWrap is set to false.

Basically what i am seeing is that a string that should be wrapping can lose the last word or words if there is a trailing space in the string.  This can be reproduced in the demo in the 11th example.

All you need to do is change the top string to:
            const wrappingText = `Global word wrap and alignment properties are controlled by the "default" style, and can\'t be overridden by other styles. `;

Basically just add a space to the end.  This results in 'other styles.' disappearing.

Additionally if you change the bottom string to:
            const nonWrappingText = `This one doesn't have \`wordWrap\` so it just keeps going and going and going off the edge! `;

Once again, simply adding a space to the end, the entire line of text disappears.

Now to fix this, I simply do a trim() on all my text before passing it in, but it feels like this should be done under the hood, but it also begs the question as to why/how a single trailing space is breaking things.

Hopefully this is enough info to test/reproduce and fix this issue.

Thanks again!

**Comment by mimshwright** [comment]
Thank you for your help on identifying this @andrewleek-droplab and for the PR. I'm going to review this now.

**Comment by mimshwright** [comment]
So far, PR #117 has not fixed both related issues (#118 and #100). I wrote some tests and did some more demos of this and haven't quite gotten it yet (although I did find at least 1 test that would have caught this but I didn't realize I was testing it wrong). In some cases, the entire final line is disappearing. It seems that the logic flow is missing some case and it's never writing the final word to the line or the final line to the paragraph. I'm stepping thru the logic again to try to find a solution.

**Comment by andrewleek-droplab** [comment]
Hey, so just to be clear, PR #117 was the first bug I found and #118 was another I found afterwards.  So while #117 might or might not fix #100, it certainly wont fix #118.  When I found the issue explain in #118 I immediately thought that perhaps that was the issue all along and even reverted #117 to test this but it did not fix things.  The only way I have things working for me (with normal text, ie not emoji's) was with #117 and then trimming the string to clear trailing spaces (#118). Then everything worked as expected.

So basically, for me, I feel #117 and #118 are separate issues/bugs. I hope that clears things up.

Let me know if you need any more info

**Comment by mimshwright** [comment]
@andrewleek-droplab Thank you for clarifying. I was confused and thought 117 was a fix for 118. But they're all somewhat related to layout and wrapping not working correctly. 

For all of these issues, I'm trying to write tests that actually cover all of these cases so we can make sure the fixes are actually working.

**Comment by mimshwright** [comment]
@andrewleek-droplab Can you check out v2.2.7? This should be working correctly in that version. Let me know if there are still issues.
